packages/noya-designsystem/src/components/InputField.tsx

```diff
@@ -200,10 +260,14 @@ function InputFieldRoot({
   labelPosition = 'end',
   labelSize = 6,
 }: InputFieldRootProps) {
-  const contextValue = useMemo(() => ({ labelPosition, labelSize }), [
-    labelPosition,
-    labelSize,
-  ]);
+  const hasDropdown = Children.toArray(children).some(
```

Comment:
I added this. We can detect `hasDropdown` automatically by checking the types of the `children` elements, which is a little more convenient